js/modules/modules.go

```diff
@@ -50,6 +50,11 @@ type VU interface {
 	// interface.
 	Events() common.Events
 
+	// TracerProvider returns the VU TracerProvider.
+	// NOTE: This API is EXPERIMENTAL and may be changed, renamed or
+	// completely removed in a later k6 release.
+	TracerProvider() common.TracerProvider
```

Comment:
Initially I added the `TracerProvider` to `VU.State` as suggested by @mstoykov , but the problem that I was facing with this, at least from the _xk6-browser_ side, is that `State` is `nil` when instantiating a new module instance (actual call to `NewModuleInstance`), where we expect to have access to the tracer provider. Considering also that `State` seems to represent the `VU`'s volatile state, and that the tracer provider is not tight to a `VU`, but is a "more global entity" I decided to expose it directly through `VU` interface. Let me know what do you think.

Reply:
From my understanding just like with metrics you will only emit traces within the actual execution of the test code.

As such I will prefer if keep all the tracing similar to metrics within the bounds of the execution of the test.